apps/prairielearn/src/components/DeleteCourseInstanceModal.tsx

```diff
@@ -0,0 +1,194 @@
+import { useEffect, useRef, useState } from 'preact/hooks';
+
+import type { CourseInstance } from '../lib/db-types.js';
+
+export function DeleteCourseInstanceModal({
+  shortName,
+  enrolledCount,
+  csrfToken,
+}: {
+  shortName: CourseInstance['short_name'];
+  enrolledCount: number;
+  csrfToken: string;
+}) {
+  const [step, setStep] = useState(1);
+  const [checks, setChecks] = useState({
+    impact: false,
+    irreversible: false,
+    recovery: false,
+  });
+  const [confirmationText, setConfirmationText] = useState('');
+
+  const allChecked = Object.values(checks).every(Boolean);
+  const canContinue = step === 1 ? allChecked : confirmationText === shortName;
+
+  const labelId = 'deleteCourseInstanceModalLabel';
+
+  // Reset state whenever the Bootstrap modal fully hides (after fade animation)
+  const modalRef = useRef<HTMLDivElement | null>(null);
```

Comment:
Why not use `react-bootstrap`? It has hooks for `onHide`

Reply:
> That's not the worst thing in the world, just not the route I chose.

I don't particularly care either way, but I always try to write Preact with as few DOM native APIs as possible. I'm happy with how this is right now, but in my own code, I will write it the other way.

Reply:
Is that a bug that should be raised with react-bootstrap? If the component library supposed to mirror bootstrap isn't doing a good job mirroring bootstrap, to the point we would prefer to not use it, that seems like an issue.

Reply:
Done: http://redirect.github.com/react-bootstrap/react-bootstrap/issues/6953